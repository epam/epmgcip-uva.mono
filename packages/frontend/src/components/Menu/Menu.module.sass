@use 'src/style/colors'
@import 'src/style/typography'

.menuWrapper
    z-index: -1
    position: absolute
    top: 0
    left: 0
    opacity: 0
    width: 100vw
    height: 100vh
    background-color: colors.$blue-dark
    padding: 8px 12px 0 30px
    box-sizing: border-box

    .menuContent
        max-width: calc(768px - 30px)
        display: flex
        flex-direction: column
        margin: 0 auto

    .menuCloseButton
        width: 30px

    .menuLink
        @extend .H3
        color: colors.$white
        padding-bottom: 24px

        &:hover
            text-decoration: underline

    .currentMenuLink
        text-decoration: underline

.modalCancelButton
    color: colors.$violet
    background-color: colors.$white
    border: 1px solid colors.$violet

    &:hover
        color: colors.$white
        background-color: colors.$blue-light
        border: none

.modalUserLogoutButton
    background-color: colors.$blue-light
    color: colors.$white

    &:hover
        background-color: colors.$blue-lightest