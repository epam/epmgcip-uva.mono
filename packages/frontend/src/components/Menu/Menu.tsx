import { Button } from 'src/components';
import css from './Menu.module.sass';
import CloseSvg from 'src/assets/cross.svg';
import { useDispatch } from 'react-redux';
import { Dispatch } from '@reduxjs/toolkit';
import { setMenu, setUserName } from 'src/redux/actions';
import { Link } from 'src/components';
import { useNavigate } from 'react-router-dom';
import {
  CREATE_EVENT_ROUTE,
  EVENTS_ROUTE,
  MANAGE_USERS_ROUTE,
  ROOT_ROUTE,
  VOLUNTEERS_ROUTE,
} from '../../constants';

const getMenuLinkClasses = (currentRoute: string, chekingRoute: string) =>
  currentRoute === chekingRoute
    ? `${css.menuLink} ${css.currentMenuLink}`
    : css.menuLink;

export const Menu = () => {
  const dispatch: Dispatch = useDispatch();
  const navigate = useNavigate();
  const currentRoute = window.location.pathname;

  const handleCloseMenu = () => {
    dispatch(setMenu(false));
  };
  const handleChangePage = (route: string) => {
    navigate(route);
    handleCloseMenu();
  };
  const handleLogout = () => {
    dispatch(setUserName(''));
    handleChangePage(ROOT_ROUTE);
  };

  return (
    <div id='menu' className={css.menuWrapper}>
      <div className={css.menuContent}>
        <Button onClick={handleCloseMenu}>
          <img className={css.menuCloseButton} src={CloseSvg} />
        </Button>
        <Link
          onClick={() => handleChangePage(EVENTS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, EVENTS_ROUTE)}
        >
          Events
        </Link>
        <Link
          onClick={() => handleChangePage(CREATE_EVENT_ROUTE)}
          className={getMenuLinkClasses(currentRoute, CREATE_EVENT_ROUTE)}
        >
          Create Event
        </Link>
        <Link
          onClick={() => handleChangePage(VOLUNTEERS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, VOLUNTEERS_ROUTE)}
        >
          Volunteers List
        </Link>
        <Link
          onClick={() => handleChangePage(MANAGE_USERS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, MANAGE_USERS_ROUTE)}
        >
          Manage Stuff
        </Link>
        <Link onClick={handleLogout} className={css.menuLink}>
          Logout
        </Link>
      </div>
    </div>
  );
};
