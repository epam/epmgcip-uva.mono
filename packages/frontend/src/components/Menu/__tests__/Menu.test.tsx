import {  describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import { Menu } from '../Menu';
import { Provider } from 'react-redux';

const observableMock = vi.fn();
const dispatchMock = <T,>() => vi.fn() as T;
const getStateMock = vi.fn();
const subscribeMock = vi.fn();
const replaceReducerMock = vi.fn();
const navigateMock = vi.fn();

const storeMock = {
  [Symbol.observable]: observableMock,
  dispatch: dispatchMock,
  getState: () => getStateMock,
  subscribe: () => subscribeMock,
  replaceReducer: () => replaceReducerMock,
};

vi.mock('react-redux', async () => {
  const actual = await vi.importActual('react-redux');

  return {
    ...actual,
    useDispach: () => ({ dispatch: dispatchMock }),
  };
});

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');

  return {
    ...actual,
    useNavigate: () => ({ navigate: navigateMock }),
  };
});

describe('Testing: Menu', () => {
  it('should render', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <Menu />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
