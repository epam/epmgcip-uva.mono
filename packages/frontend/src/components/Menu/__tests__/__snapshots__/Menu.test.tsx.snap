// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Menu > should render 1`] = `
<div>
  <div
    class="_menuWrapper_5677b1"
    id="menu"
  >
    <div
      class="_menuContent_5677b1"
    >
      <button
        class="_button_6e8901"
      >
        <img
          class="_menuCloseButton_5677b1"
          src="/packages/frontend/src/assets/cross.svg"
        />
      </button>
      <a
        class="_link_04e8aa _menuLink_5677b1"
      >
        Events
      </a>
      <a
        class="_link_04e8aa _menuLink_5677b1"
      >
        Create Event
      </a>
      <a
        class="_link_04e8aa _menuLink_5677b1"
      >
        Volunteers List
      </a>
      <a
        class="_link_04e8aa _menuLink_5677b1"
      >
        Manage Stuff
      </a>
      <a
        class="_link_04e8aa _menuLink_5677b1"
      >
        Logout
      </a>
    </div>
  </div>
</div>
`;
